cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/Authenticator.java

```diff
@@ -16,8 +16,9 @@
      *
      * @param cloudContext    the context containing information to identify which stack (cluster) is affected
      * @param cloudCredential credentials for the given platform
+     * @param verification    authentication is used to verify credentials during creating credentials
      * @return authenticated context object, shall contain a the authenticated client object. The client object needs to be threadsafe.
      */
-    AuthenticatedContext authenticate(CloudContext cloudContext, CloudCredential cloudCredential);
+    AuthenticatedContext authenticate(CloudContext cloudContext, CloudCredential cloudCredential, boolean verification);
```

Comment:
Am I missing something, because I don't see any usage of this boolean flag?

Reply:
The only place where it is true is:
CredentialVerificationHandler:L42

Reply:
I see that, but I don't see it beeing used (like this parameter: firstTime)

Reply:
Wouldn't it be better to have 2 authenticate method in the interface? 
One with the verification boolean, and one without. 
This way only the CredentialVerificationHandler:L42 would use it, and the other classes could go untouched.
Default implementation in the interface could refer to the original with false?